File: 백준/Silver/1789. 수들의 합/수들의 합.cc
#include <iostream>

using namespace std;

int main(void) {
    long long n;
    cin >> n;
    
    long long sum = 0;
    int result = 0;
    for (long long i = 1; ; i++) {
        sum = sum + i;
        result++;
        if (sum > n) {
            result--;
            cout << result << '\n';
            return 0;
        }
    }
}

File: 백준/Silver/1789. 수들의 합/수들의 합.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        long n = scanner.nextLong();
        
        long sum = 0;
        int result = 0;
        for (int i = 1; ; i++) {
            if (sum > n) {
                result--;
                break;
            } else {
                sum += i;
                result++;
            }
        }
        
        System.out.println(result);
    }
}